demos/projects/ST/b-l475e-iot01a/port/azure_iot_flash_platform.c

```diff
@@ -37,72 +44,197 @@ static AzureIoTResult_t prvBase64Decode( uint8_t * base64Encoded,
+    {
+        ulReadSize = pxAduImage->ulImageFileSize - ulOffset < sizeof( ucPartitionReadBuffer ) ? pxAduImage->ulImageFileSize - ulOffset : sizeof( ucPartitionReadBuffer );
 
-    LogInfo( ( "AzureIoTPlatform_EnableImage()\r\n" ) );
+        memcpy( ucPartitionReadBuffer, ( pxAduImage->xUpdatePartition + ulOffset ), ulReadSize );
 
-    return eAzureIoTSuccess;
+        mbedtls_md_update( &ctx, ( const unsigned char * ) ucPartitionReadBuffer, ulReadSize );
+
+        if( ( ulOffset % 65536 == 0 ) && ( ulOffset != 0 ) )
+        {
+            printf( "." );
+        }
+    }
+
+    printf( "\r\n" );
+
+    AZLogInfo( ( "Done\r\n" ) );
+
+    mbedtls_md_finish( &ctx, ucCalculatedHash );
+    mbedtls_md_free( &ctx );
+
+    if( memcmp( ucDecodedManifestHash, ucCalculatedHash, azureiotflashSHA_256_SIZE ) == 0 )
+    {
+        AZLogInfo( ( "SHAs match\r\n" ) );
+        xResult = eAzureIoTSuccess;
+    }
+    else
+    {
+        AZLogInfo( ( "SHAs do not match\r\n" ) );
+        AZLogInfo( ( "Wanted: " ) );
+
+        for( int i = 0; i < azureiotflashSHA_256_SIZE; ++i )
+        {
+            printf( "%x", ucDecodedManifestHash[ i ] );
+        }
+
+        printf( ( "\r\n" ) );
+        AZLogInfo( ( "Calculated: " ) );
+
+        for( int i = 0; i < azureiotflashSHA_256_SIZE; ++i )
+        {
+            printf( "%x", ucCalculatedHash[ i ] );
+        }
+
+        printf( ( "\r\n" ) );
+
+        xResult = eAzureIoTErrorFailed;
+    }
+
+    return xResult;
 }
 
-AzureIoTResult_t AzureIoTPlatform_ResetDevice( AzureADUImage_t * const pxAduImage )
+AzureIoTResult_t AzureIoTPlatform_EnableImage( AzureADUImage_t * const pxAduImage )
 {
-    (void)pxAduImage;
-
-    // TODO: Fill in to reboot the device.
+    FLASH_OBProgramInitTypeDef xOptionBytes;
+    int xResult = eAzureIoTSuccess;
+
+    HAL_FLASH_Unlock();
+    HAL_FLASH_OB_Unlock();
+    HAL_FLASHEx_OBGetConfig( &xOptionBytes ); /* Get current optionbytes configuration */
+    xOptionBytes.OptionType = OPTIONBYTE_USER;
+    xOptionBytes.USERType = OB_USER_BFB2;
+    /* If BFB2 set, we will reset it when we reboot the device */
+    xOptionBytes.USERConfig =
+        ( ( xOptionBytes.USERConfig & OB_BFB2_ENABLE ) == OB_BFB2_ENABLE )
+        ? OB_BFB2_DISABLE
+        : OB_BFB2_ENABLE;
+
+    if( HAL_FLASHEx_OBProgram( &xOptionBytes ) != HAL_OK )
+    {
+        AZLogError( ( "Error setting option bytes\r\n" ) );
+        xResult = eAzureIoTErrorFailed;
+    }
 
-    LogInfo( ( "AzureIoTPlatform_ResetDevice()\r\n" ) );
+    HAL_FLASH_Lock();
+    HAL_FLASH_OB_Lock();
+    return xResult;
+}
 
+AzureIoTResult_t AzureIoTPlatform_ResetDevice( AzureADUImage_t * const pxAduImage )
+{
+    HAL_FLASH_Unlock();
+    HAL_FLASH_OB_Unlock();
+
+    /*
+     * Sets options bits and restarts device.
+     * Also sets the boot bank address to 0x08000000, which means we always write
+     * to 0x08080000
+     */
+    HAL_FLASH_OB_Launch();
     return eAzureIoTSuccess;
 }
```

Comment:
I don't know if it's part of the code style script or not but we try and give some room to logical blocks. Assignments can be grouped together but then return values, `if` blocks etc get a new line around them

Reply:
It adds new lines around assignments at the top of a function and if statements, but not return values! I'll do a scan for this across the PR